src/arch/arm/c_traps.c

```diff
@@ -222,7 +231,16 @@ VISIBLE NORETURN void c_handle_vcpu_fault(word_t hsr)
     ksKernelEntry.path = Entry_VCPUFault;
     ksKernelEntry.word = hsr;
 #endif
-    handleVCPUFault(hsr);
+
+#ifdef CONFIG_HARDWARE_DEBUG_API
+    word_t esr = getESR();
+    if (isDebugFault(esr)) {
+        handleDebugFaultEvent(esr);
+    } else
+#endif
+    {
+        handleVCPUFault(hsr);
+    }
```

Comment:
Currently `armv_handleVCPUFault()` already checks for FPU faults and user level faults, it would be more consistent to have the debug check there too.